easy_phi/static/js/main.js

```diff
@@ -159,21 +145,17 @@ var ep = window['ep'] || {
         // create containers for modules by number of slots in this platform
         var module_list_container = $('#modules_list');
         module_list_container.empty();
-        // ep.slots + 1 because of Broadcast module in slot 0
-        for (var i=0; i<ep.slots+1; i++)
-            ep._add_module(module_list_container, i, ep._empty_slot_str);
 
         // get list of modules and update created containers
         $.get(ep.base_url+"/api/v1/modules_list?format=json", function(modules){
             modules.forEach(function(module_name, slot_id) {
-                if (slot_id > ep.slots) {
-                    /* module plugged through standalone adapter or platform
-                    * ports are not configured - add new slot dynamically */
-                    ep._add_module(module_list_container, slot_id, ep._empty_slot_str);
-                }
-
+                ep._add_module(module_list_container, slot_id, ep._empty_slot_str);
                 ep._updateModuleUI(slot_id, module_name);
             });
+
+            // Websocket handler assigned after module list updated manually to
+            // evade race condition
+            ep._ws.onmessage = ep._parseWSMessage;
```

Comment:
websocket handler assigned after all modules and slot containers created to evade race condition